node_modules/col-lti/tests/test-lti.js

```diff
@@ -13,18 +13,172 @@
+                return callback();
+              });
+            });
+          });
+        });
+      });
+
+      /**
+       * Test that verifies that new courses are created on the fly
+       */
+      it('creates courses on the fly', function(callback) {
+        // Launch a user into a new course
+        LtiTestsUtil.getAssetLibraryClient(null, null, null, function(client, courseA, userA) {
+
+          // Get the user's information
+          UsersTestsUtil.assertGetMe(client, courseA, function(me) {
+            assert.strictEqual(me.canvas_user_id, userA.id);
+            assert.strictEqual(me.course.canvas_course_id, courseA.id);
+
+            // Launching the tool for the same user in the same course should
+            // not result in a new course
+            LtiTestsUtil.getAssetLibraryClient(null, courseA, userA, function(client) {
+              UsersTestsUtil.assertGetMe(client, courseA, function(meReload) {
+                assert.strictEqual(meReload.course.id, me.course.id);
+
+                // Launching the tool for a new user in the same course should
+                // not result in a new Collabosphere course
+                LtiTestsUtil.getAssetLibraryClient(null, courseA, null, function(client, courseA, userB) {
+                  UsersTestsUtil.assertGetMe(client, courseA, function(meNewUser) {
+                    assert.strictEqual(meNewUser.course.id, me.course.id);
+
+                    // Launching the tool for the same user in another course
+                    // should result in a new Collabosphere course
+                    LtiTestsUtil.getAssetLibraryClient(null, null, userA, function(client, courseB, userA) {
+                      UsersTestsUtil.assertGetMe(client, courseB, function(meNewCourse) {
+                        assert.notEqual(meNewCourse.course.id, me.course.id);
+                        return callback();
+                      });
+                    });
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+
+      /**
+       * Test that verifies that new users are created on the fly
+       */
+      it('creates users on the fly', function(callback) {
+        // Launch a user into a new course
+        LtiTestsUtil.getAssetLibraryClient(null, null, null, function(client, courseA, userA) {
+
+          // Get the user's information
+          UsersTestsUtil.assertGetMe(client, courseA, function(me) {
+            assert.strictEqual(me.canvas_user_id, userA.id);
+            assert.strictEqual(me.course.canvas_course_id, courseA.id);
+
+            // Launching the tool for the same user in the same course should
+            // authenticate the client to the same Collabosphere user
+            LtiTestsUtil.getAssetLibraryClient(null, courseA, userA, function(client) {
+              UsersTestsUtil.assertGetMe(client, courseA, function(meReload) {
```

Comment:
Should we optionally pass in an expected `me` object to allow us to compare the different properties?